utils/file-utils.js

```diff
@@ -0,0 +1,70 @@
+const fileType = require('file-type');
+const Stream = require('stream');
+const request = require('request-promise');
+const fs = require('fs');
+
+exports.file = (p) => {
+  if (exports.isUrl(p)) {
+    return request.get({ url: p, encoding: null });
+  }
+  return fs.createReadStream(p);
+};
+
+exports.isUrl = (s) => {
+  const regexp = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-/]))?/;
```

Comment:
Can you use http://nodejs.org/dist/latest/docs/api/url.html#url_class_url instead for this? Attempt to construct a `new URL` and add a try..catch to see if it succeeds?